src/include/platform/internal/GenericConnectivityManagerImpl_WiFi.h

```diff
@@ -194,6 +198,24 @@ inline CHIP_ERROR GenericConnectivityManagerImpl_WiFi<ImplClass>::_SetPollingInt
 }
 #endif
 
+template <class ImplClass>
+inline CHIP_ERROR GenericConnectivityManagerImpl_WiFi<ImplClass>::_GetNetworkSSID(MutableCharSpan & ssid)
```

Comment:
So are there any implementations of these bits?

Reply:
I will push a separate commit for NXP platforms after I finish up on the Matter generic code.

Reply:
@bzbarsky-apple and @Damian-Nordic , can you please review the latest push?
I am trying to fix the NRF issue from https://github.com/project-chip/connectedhomeip/actions/runs/8523124172/job/23344748833?pr=31684 caused by the change but I am unable to build the project using 
`./scripts/examples/nrfconnect_example.sh all-clusters-app nrf7002dk_nrf5340_cpuapp -DCONF_FILE=prj_release.conf`. I have an error regarding west:
`_=/usr/bin/env
+ west build -p auto -b nrf7002dk_nrf5340_cpuapp -d all-clusters-app/nrfconnect/build all-clusters-app/nrfconnect -- -DCONFIG_CHIP_DEBUG_SYMBOLS=n -DCONF_FILE=prj_release.conf
usage: west [-h] [-z ZEPHYR_BASE] [-v] [-V] <command> ...
west: error: argument <command>: invalid choice: 'build' (choose from 'init', 'update', 'list', 'manifest', 'diff', 'status', 'forall', 'help', 'config', 'topdir', 'selfupdate')`

Can you please advise?
Thanks.

Reply:
@mihai-ignat You get the `west` issue because you likely not have nRF Connect SDK installed and `west` cannot find Zephyr project that defines the `build` command. But CI error is caused by `ConnectivityManagerImpl` on our platform not inheriting from `ConnectivityManager_Wifi`. You could add those dummy implementations to `GenericConnectivityManager` (without `_Wifi` suffix) to fix this.

But my main question is, do you actually need those `ConnectivityManager` changes at all? Couldn't the shell commands talk to the `WiFiDriver` alone? That way the shell commands would use what the real code does and you wouldn't need to provide platform API changes that would be used by shell only and likely not be implemented by all the platforms.

Reply:
@Damian-Nordic Adding the extra features required these functions that were not available in the WiFiDriver or the ConnectivityManager. I had a previous version with WiFiDriver implementing the functions and that impacted all platform files so that's why I switched to ConnectivityManager. If you have any ideas on how to minimize the impact on the platforms, can you please share? Or I can drop the PR if the complexity of the WiFi shell is too much and just keep the old "connect" command. Thanks.

Reply:
@mihai
I think it's better that shell commands use existing APIs instead of adding alternative ones.

To that end I suggest:
- remove ConnectivityManager's `GetNetworkSSID` - after all there can be multiple networks stored on the device so the API is not generic enough
- remove ConnectivityManager's `GetNetworkPassword` - same reason as above + it's unused now.
- make "connect" shell command take "ssid" as an argument.
- `DisconnectNetwork` is fine with me since there's no equivalent API in `WiFiDriver`. Well, there's existing `SetWiFiStationMode` but it's undocumented and all platforms implement it differently :)

Would that be OK with you?

Reply:
Hello, @Damian-Nordic.
I have pushed a commit to simplify the changes, keeping what you suggested. I also tried fixing the Nordic build, awaiting the build result.
Hope this is simple enough for a generic usage of wifi shell and for further features, each platform vendors can improve the shell using custom commands in their own apps.
BR,
Mihai